csharp/src/Google.Protobuf/MessageExtensions.cs

```diff
@@ -253,37 +253,21 @@ internal static void MergeFrom(this IMessage message, byte[] data, bool discardU
         {
             ProtoPreconditions.CheckNotNull(message, nameof(message));
             ProtoPreconditions.CheckNotNull(data, nameof(data));
-            CodedInputStream input = new CodedInputStream(data)
-            {
-                DiscardUnknownFields = discardUnknownFields,
-                ExtensionRegistry = registry
-            };
-            message.MergeFrom(input);
-            input.CheckReadEndOfStreamTag();
+            MergeFrom(message, new ReadOnlySpan<byte>(data), discardUnknownFields, registry);
```

Comment:
The three internal methods you've changed here are each only called from a single place (earlier in the same file). Please could you inline them there, and then just remove these internal methods?

(I've done this locally, and could just push another commit if you'd like me to.)